File: TypeScript/TP3_MET_TS/src/Inductif/CaractereMotInductif.ts
import Mot from "../Interface/Mot";
import {MotVideInductif} from "./MotVideInductif";
import {MotMotInductif} from "./MotMotInductif";


export class CaractereMotInductif extends Mot {

    public static FAB: Mot = MotVideInductif.SINGLETON;

    private _caractere: string;
    private _reste: Mot;

    public constructor(caractere: string, reste: Mot) {
        super();
        this._caractere = caractere;
        this._reste = reste;
    }

    public caractere(): string {
        return this._caractere;
    }

    public reste(): Mot {
        return this._reste;
    }

    public creerMotVide(): Mot {
        return MotVideInductif.SINGLETON;
    }

    public creerCaractereMot(c: string): Mot {
        return new CaractereMotInductif(c, this);
    }

    public creerMotMot(droite: Mot): Mot {
        return new MotMotInductif(this, droite);
    }


    public casCaractereMot(): boolean {
        return true;
    }

}

File: Java/TP3_MET_JAVA/src/Inductif/CaractereMotInductif.java
package Inductif;

import Interface.Mot;

public class CaractereMotInductif implements Mot {


    public static final Mot FAB = MotVideInductif.SINGLETON;

    private Character caractere;
    private Mot reste;

    public CaractereMotInductif(Character caractere, Mot reste) {
        this.caractere = caractere;
        this.reste = reste;
    }

    public Character caractere() {
        return caractere;
    }

    @Override
    public Mot reste() {
        return reste;
    }

    @Override
    public Mot creerMotVide() {
        return MotVideInductif.SINGLETON;
    }

    @Override
    public Mot creerCaractereMot(char c) {
        return new CaractereMotInductif(c, this);
    }

    @Override
    public Mot creerMotMot(Mot droite) {
        return new MotMotInductif(this, droite);
    }


    public boolean casCaractereMot() {
        return true;
    }

}
